### PR TITLE
Deprecate tuple enrichments. Fixes #2763

### LINKED ISSUE #2763: Deprecate the tuple map enrichments

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #2763

### CODE REVIEW

File: main/settings/src/main/scala/sbt/Structure.scala [hunk 1]
```diff
@@ -304,20 +304,16 @@ object Scoped {
...
-	implicit def t14ToTable14[A,B,C,D,E,F,G,H,I,J,K,L,N,O](t14: (ScopedTaskable[A], ScopedTaskable[B], ScopedTaskable[C], ScopedTaskable[D], ScopedTaskable[E], ScopedTaskable[F], ScopedTaskable[G], ScopedTaskable[H], ScopedTaskable[I], ScopedTaskable[J], ScopedTaskable[K], ScopedTaskable[L], ScopedTaskable[N], ScopedTaskable[O]) ): RichTaskable14[A,B,C,D,E,F,G,H,I,J,K,L,N,O] = new RichTaskable14(t14)
-	implicit def t15ToTable15[A,B,C,D,E,F,G,H,I,J,K,L,N,O,P](t15: (ScopedTaskable[A], ScopedTaskable[B], ScopedTaskable[C], ScopedTaskable[D], ScopedTaskable[E], ScopedTaskable[F], ScopedTaskable[G], ScopedTaskable[H], ScopedTaskable[I], ScopedTaskable[J], ScopedTaskable[K], ScopedTaskable[L], ScopedTaskable[N], ScopedTaskable[O], ScopedTaskable[P]) ): RichTaskable15[A,B,C,D,E,F,G,H,I,J,K,L,N,O,P] = new RichTaskable15(t15)*/
+  @deprecated("The tuple enrichments have been deprecated and will be removed in sbt 1.0.0", "0.13.13") implicit def t2ToTable2[A, B](t2: (ScopedTaskable[A], ScopedTaskable[B])): RichTaskable2[A, B] = new RichTaskable2(t2)
```
Comment (eed3si9n) [inline main/settings/src/main/scala/sbt/Structure.scala]: I think we need a better wording on this. Probably a link to a migration document or something.

**Review by eed3si9n — CHANGES_REQUESTED** [review]

File: main/settings/src/main/scala/sbt/Structure.scala (hunk 1 above)
Comment (dwijnand) [reply main/settings/src/main/scala/sbt/Structure.scala]: Such as?

**Comment by mtk** [comment]
I l was 88 su 79 tr w mom  e eo

File: main/settings/src/main/scala/sbt/Structure.scala (hunk 1 above)
Comment (eed3si9n) [reply main/settings/src/main/scala/sbt/Structure.scala]: sbt 0.10 style DSL is deprecated. `(k1, k2) map { (x, y) => ... }` should now be `{ val x = k1.value; val y = k2.value }`. See scala-sbt.org/blabla for more details.

**Comment by dwijnand** [comment]
will migrate to linking to the end URL of https://github.com/sbt/website/pull/284 once that's reviewed and merged.

**Comment by eed3si9n** [comment]
The doc is now merged, so could you update the link?

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**